worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,222 @@
+
+        create_region(multiworld, player, locations_per_region, location_cache, "4-1"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Marching Milde's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Hookbill The Koopa's Boss Room"),
+
+        create_region(multiworld, player, locations_per_region, location_cache, "5-1"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Sluggy The Unshaven's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Raphael The Raven's Boss Room"),
+
+        create_region(multiworld, player, locations_per_region, location_cache, "6-1"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Tap-Tap The Red Nose's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Bowser's Room"),
+
+    ]
+    if world.options.extras_enabled.value == 1:
+        regions.insert(68, create_region(multiworld, player, locations_per_region, location_cache, "6-Extra"))

```

Comment:
Do you really think it's a good idea to add these extra levels at magic-numbered indexes like this?

Reply:
Because of how region creation is done, I wanted the Extra levels to follow the in-game placement order. Because they were only created if the setting was on, though, it had to create them after every other level (Unless I wanted to add 6 stops in the region creation to check the Extra level for that world)

Reply:
I think that will be harder to maintain, but it's not a hill to die on. I said what I had to say, and after that, you're the boss :)